pkg/transforms/conversion.go

```diff
@@ -73,3 +77,55 @@ func (f Conversion) TransformToJSON(edgexcontext *appcontext.Context, params ...
 	}
 	return false, errors.New("Unexpected type received")
 }
+
+// TransformToCloudEvent will transform a models.Event to a Cloud Event
+// It will return an error and stop the pipeline if a non-edgex event is received or if no data is received.
+func (f Conversion) TransformToCloudEvent(edgexcontext *appcontext.Context, params ...interface{}) (continuePipeline bool, stringType interface{}) {
+	if len(params) < 1 {
+		return false, errors.New("No Event Received")
+	}
+	edgexcontext.LoggingClient.Debug("Transforming to CloudEvent")
+	if result, ok := params[0].(models.Event); ok {
+		if len(result.Readings) == 0 {
+			return false, errors.New("No event readings to transform")
+		}
+
+		source := result.Device
+		event := cloudevents.NewEvent(cloudevents.VersionV1)
+		event.SetID(result.ID)
+		event.SetType(reflect.TypeOf(result).String())
+		event.SetSource(source)
+		// The whole event containing all of the readings is serialized into the cloud event.
+		if err := event.SetData(result); err != nil {
```

Comment:
So we are just wrapping the EdgeX event in a Cloud Event???
Not sure how valuable that will be as receivers will still deal with an EdgeX event once un-wrapped.

I think we need to do an actual conversion from Reading to CloudEvent. So result for function is a slice. There reading Value goes into event.SetData().  If I remember correctly there are attributes that can be set for the remaining metadata.